src/cdk-cli-integ-tests.ts

```diff
@@ -0,0 +1,292 @@
+        MAVEN_ARGS: '--no-transfer-progress',
+        // This is not actually a canary, but this prevents the tests from making
+        // assumptions about the availability of source packages.
+        IS_CANARY: 'true',
+        CI: 'true',
+      },
+      strategy: {
+        failFast: false,
+        matrix: {
+          domain: {
+            suite: [
+              'cli-integ-tests',
+              'init-csharp',
+              'init-fsharp',
+              'init-go',
+              'init-java',
+              'init-javascript',
+              'init-python',
+              'init-typescript-app',
+              'init-typescript-lib',
+              'tool-integrations',
+            ],
+          },
+        },
+      },
+      steps: [
+        {
+          name: 'Download build artifacts',
+          uses: 'actions/download-artifact@v4',
+          with: {
+            name: 'build-artifact',
+            path: 'packages',
+          },
+        },
+        {
+          name: 'Set up JDK 18',
+          if: 'matrix.suite == \'init-java\' || matrix.suite == \'cli-integ-tests\'',
+          uses: 'actions/setup-java@v4',
+          with: {
+            'java-version': '18',
+            'distribution': 'corretto',
+          },
+        },
+        {
+          name: 'Authenticate Via OIDC Role',
+          id: 'creds',
+          uses: 'aws-actions/configure-aws-credentials@v4',
+          with: {
+            'aws-region': 'us-east-1',
+            'role-duration-seconds': 4 * 60 * 60,
+            // Expect this in Environment Variables
+            'role-to-assume': '${{ vars.AWS_ROLE_TO_ASSUME_FOR_TESTING }}',
+            'role-session-name': 'run-tests@aws-cdk-cli-integ',
+            'output-credentials': true,
+          },
+        },
+        // This is necessary for the init tests to succeed, they set up a git repo.
+        {
+          name: 'Set git identity',
+          run: [
+            'git config --global user.name "aws-cdk-cli-integ"',
+            'git config --global user.email "noreply@example.com"',
+          ].join('\n'),
+        },
+        {
+          name: 'Install Verdaccio',
+          run: 'npm install -g verdaccio pm2',
+        },
+        {
+          name: 'Create Verdaccio config',
+          run: [
+            'mkdir -p $HOME/.config/verdaccio',
+            `echo '${JSON.stringify(verdaccioConfig)}' > $HOME/.config/verdaccio/config.yaml`,
+          ].join('\n'),
+        },
+        {
+          name: 'Start Verdaccio',
+          run: [
+            'pm2 start verdaccio -- --config $HOME/.config/verdaccio/config.yaml',
+            'sleep 5 # Wait for Verdaccio to start',
+          ].join('\n'),
+        },
+        {
+          name: 'Configure npm to use local registry',
+          run: [
+            'npm config set registry http://localhost:4873/',
+            'echo \'//localhost:4873/:_authToken="MWRjNDU3OTE1NTljYWUyOTFkMWJkOGUyYTIwZWMwNTI6YTgwZjkyNDE0NzgwYWQzNQ=="\' > ~/.npmrc',
```

Comment:
Can you add a note about this token? it looks scary.

Reply:
Explained in comments